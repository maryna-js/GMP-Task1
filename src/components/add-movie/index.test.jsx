import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import AddMovieModal from "./index";
import { render, fireEvent, screen, waitFor } from "@testing-library/react";

const mockStore = configureMockStore();
const store = mockStore({});

describe("AddMovieModal", () => {
    const mockCreateMovieData = jest.fn();
    const mockShowAddMovie = jest.fn();
    const mockCloseAddMovieModal = jest.fn();

    const renderModalForm = () =>
        render(
            <Provider store={store}>
                <AddMovieModal
                    createMovieData={mockCreateMovieData}
                    showAddMovie={mockShowAddMovie}
                    closeAddMovieModal={mockCloseAddMovieModal}
                />
            </Provider>
        );

    it("can render correctly", () => {
        const { asFragment } = renderModalForm();
        expect(asFragment()).toMatchSnapshot();
    });

    it("if empty form will not call save", () => {
        renderModalForm();
        fireEvent.click(screen.getByText("Save"));
        expect(mockCreateMovieData).not.toBeCalled();
    });
});

