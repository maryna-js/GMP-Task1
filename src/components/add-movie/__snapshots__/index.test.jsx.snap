// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovieModal can render correctly 1`] = `
<DocumentFragment>
  <div
    class="add-modal"
  >
    <div
      class="add-modal-header"
    >
      <button
        class="add-modal-close"
      >
        X
      </button>
    </div>
    <div>
      <h2
        class="add-modal-heading"
      >
        Add movie
      </h2>
    </div>
    <form
      action="#"
    >
      <div
        class="add-modal-field-container"
      >
        <label
          class="add-modal-field-label"
          for="title"
        >
          Title *
        </label>
        <input
          class="add-modal-field-input"
          name="title"
          placeholder="Title"
          type="text"
          value=""
        />
      </div>
      <div
        class="add-modal-field-container"
      >
        <label
          class="add-modal-field-label"
          for="tagline"
        >
          Tagline *
        </label>
        <input
          class="add-modal-field-input"
          name="tagline"
          placeholder="tagline"
          type="text"
          value=""
        />
      </div>
      <div
        class="add-modal-field-container"
      >
        <label
          class="add-modal-field-label"
          for="vote_average"
        >
          Vote Average
        </label>
        <input
          class="add-modal-field-input"
          name="vote_average"
          placeholder="Vote Average"
          type="number"
          value="0"
        />
      </div>
      <div
        class="add-modal-field-container"
      >
        <label
          class="add-modal-field-label"
          for="vote_count"
        >
          Vote Count
        </label>
        <input
          class="add-modal-field-input"
          name="vote_count"
          placeholder="Vote Count"
          type="number"
          value="0"
        />
      </div>
      <div
        class="add-modal-field-container"
      >
        <label
          class="add-modal-field-label"
          for="release_date"
        >
          Release date *
        </label>
        <input
          class="add-modal-field-input"
          name="release_date"
          placeholder="Release date"
          type="date"
          value=""
        />
      </div>
      <div
        class="add-modal-field-container"
      >
        <label
          class="add-modal-field-label"
          for="poster_path"
        >
          Movie URL *
        </label>
        <input
          class="add-modal-field-input"
          name="poster_path"
          placeholder="Movie URL"
          type="text"
          value=""
        />
      </div>
      <div
        class="add-modal-field-container"
      >
        <label
          class="add-modal-field-label"
          for="genres"
        >
          Genre *
        </label>
        <select
          class="add-modal-field-input"
          multiple=""
          name="genres"
        >
          <option
            value=""
          >
            Select a genre
          </option>
          <option
            value="Comedy"
          >
            Comedy
          </option>
          <option
            value="Drama"
          >
            Drama
          </option>
          <option
            value="Horror"
          >
            Horror
          </option>
        </select>
      </div>
      <div
        class="add-modal-field-container"
      >
        <label
          class="add-modal-field-label"
          for="overview"
        >
          Overview *
        </label>
        <input
          class="add-modal-field-input"
          name="overview"
          placeholder="Overview"
          type="text"
          value=""
        />
      </div>
      <div
        class="add-modal-field-container"
      >
        <label
          class="add-modal-field-label"
          for="runtime"
        >
          Runtime *
        </label>
        <input
          class="add-modal-field-input"
          name="runtime"
          placeholder="Runtime"
          type="number"
          value="0"
        />
      </div>
      <div
        class="add-modal-field-container"
      >
        <label
          class="add-modal-field-label"
          for="budget"
        >
          Budget
        </label>
        <input
          class="add-modal-field-input"
          name="budget"
          placeholder="Budget"
          type="number"
          value="0"
        />
      </div>
      <div
        class="add-modal-field-container"
      >
        <label
          class="add-modal-field-label"
          for="revenue"
        >
          Revenue
        </label>
        <input
          class="add-modal-field-input"
          name="revenue"
          placeholder="Revenue"
          type="number"
          value="0"
        />
      </div>
      <div
        class="add-modal-button-container"
      >
        <button
          class="add-modal-button-reset"
          type="reset"
        >
          Reset
        </button>
        <button
          class="add-modal-button-submit"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
